lib/rust/ensogl/core/src/animation/frp/animation/overshoot.rs

```diff
@@ -0,0 +1,134 @@
+//! bounds. The sensitivity for setting the value past the bounds is reduced and value is limited to
+//! prevent the overshoot from being too large or too jumpy.
+
+use crate::prelude::*;
+
+use crate::animation::delayed::DelayedAnimation;
+use crate::animation::linear_interpolation;
+use crate::Animation;
+
+use enso_frp as frp;
+
+
+
+// ==================
+// === Constants  ===
+// ==================
+
+/// Determines how much of out-of-bounds portion of value change is actually applied to the
+/// animation target value. Larger values will make the overshoot more noticeable.
+const OUT_OF_BOUNDS_SENSITIVITY: f32 = 0.3;
+
+
+// ===========
+// === Frp ===
+// ===========
+
+crate::define_endpoints! {
+    Input {
+        /// Apply relative change to target value. Allow the value to overshoot.
+        soft_change_by (f32),
+        /// Apply relative change to target value. Remove any overshoot.
+        hard_change_by (f32),
+        /// Apply absolute change to target value. Allow the value to overshoot.
+        soft_change_to (f32),
+        /// Apply absolute change to target value, Remove any overshoot.
+        hard_change_to (f32),
+        /// Set the minimum bound of the target value. Values smaller than set bound will be
+        /// considered as overshoot and will bounce back.
+        set_min_bound (f32),
+        /// Set the maximum bound of the target value. Values larger than set bound will be
+        /// considered as overshoot will bounce back.
+        set_max_bound (f32),
+        /// Set the absolute maximum overshoot amplitude that the target value can have. If the
+        /// value overshoots more than set limit, it will be immediately clamped to that limit.
+        set_overshoot_limit (f32),
+        /// Set the time interval from when soft value change is applied to when bounce back
+        /// animation starts.
+        set_bounce_delay (f32),
+        /// Skip the animation and set the target value to the current value.
+        skip (),
+    }
+    Output {
+        /// The current value of the animation.
+        value (f32),
+        /// Current target value of the animation.
+        target (f32),
+    }
+}
+
+
+
+// ==========================
+// === OvershootAnimation ===
+// ==========================
+
+/// An [`Animation`] wrapper that allows the value to be smoothly manipulated within set bounds,
+/// overshoot set bounds and bounce back.
+#[derive(Clone, CloneRef, Debug, Shrinkwrap)]
+pub struct OvershootAnimation {
+    /// Public FRP api.
+    pub frp: FrpEndpoints,
+}
+
+impl OvershootAnimation {
+    /// Constructor. Hooks into preexisting network. Created `OvershootAnimation` struct does not
+    /// need to be persisted, all created FRP nodes will be managed by the passed-in network.
+    pub fn new(network: &frp::Network) -> Self {
+        let frp = Frp::extend(network);
+        // let out = &frp.private().output; // version for `define_endpoints_2`
```

Comment:
If this stays in the code, it should be annotated with a proper TODO pointing to a pivotal issue where it is resolved.